src/layer/vector/Polyline.js

```diff
@@ -103,8 +103,13 @@ L.Polyline = L.Path.extend({
 	},
 
 	_setLatLngs: function (latlngs) {
-		this._bounds = new L.LatLngBounds();
-		this._latlngs = this._convertLatLngs(latlngs);
+		this._bounds = new L.Bounds();
+		this._pxBounds = new L.Bounds(new L.Point(), new L.Point());
+		this._latlngs = latlngs;
+		this._points = [];
+		this._rings = [];
+		var unitScale = L.CRS.EPSG3857.scale(0);
```

Comment:
I tried but unfortunately `this._map` is `undefined` when `_setLatLngs` gets called. Like I said this is just a proof of concept, code isn't ready.